### ISSUE TITLE
Build fails after the initial build

### ISSUE THREAD

**Issue by anthonycastelli** [realm/realm-cocoa#3959]
## Goals
Build the project without having to run a clean

## Expected Results
Build to Succeed 

## Actual Results
Build failed with these errors:
```
Executable is /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/Realm.framework/Realm
Extracting x86_64 from Realm
fatal error: lipo: can't open input file: /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/Realm.framework/Realm (No such file or directory)
Merging extracted architectures: x86_64
fatal error: lipo: can't open input file: /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/Realm.framework/Realm-x86_64 (No such file or directory)
rm: /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/Realm.framework/Realm-x86_64: No such file or directory
Replacing original executable with thinned version
rm: /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/Realm.framework/Realm: No such file or directory
mv: rename /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/Realm.framework/Realm-merged to /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/Realm.framework/Realm: No such file or directory
Executable is /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/RealmSwift.framework/RealmSwift
Extracting x86_64 from RealmSwift
fatal error: lipo: can't open input file: /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/RealmSwift.framework/RealmSwift (No such file or directory)
Merging extracted architectures: x86_64
fatal error: lipo: can't open input file: /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/RealmSwift.framework/RealmSwift-x86_64 (No such file or directory)
rm: /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/RealmSwift.framework/RealmSwift-x86_64: No such file or directory
Replacing original executable with thinned version
rm: /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/RealmSwift.framework/RealmSwift: No such file or directory
mv: rename /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/RealmSwift.framework/RealmSwift-merged to /Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/RealmSwift.framework/RealmSwift: No such file or directory

```
## Steps to Reproduce
Add Realm as a cocoa pod and build the project

## Code Sample
Build errors

## Version of Realm and Tooling

Realm version: Master Branch
Xcode version: 8.0 Beta 5
iOS/OSX version: iOS 10/Swift 3
Dependency manager + version: Cocoapods v1.0.1

**Comment by bdash** [comment]
@anthonycastelli, can you please share more of your build log? The portion of the output you've shared seems likely to be the result of earlier errors.

**Comment by anthonycastelli** [comment]
I've upgraded my project for the new Swift 3 Beta 6 changes so when RealmSwift is updated to support that, i'll see if there were any additional logs. If I recall correctly that was it.

**Comment by bdash** [comment]
Ok. If you're seeing problems building with Xcode 8 beta 6 after #3965 is merged, please file a new issue.

**Comment by anthonycastelli** [comment]
More details for the build log
https://gist.github.com/anthonycastelli/e50fcb0926f44bf8a930e107f85a0727

(Under the correct issue this time)

**Comment by bdash** [comment]
Can you share the full build log? As I mentioned earlier, the error messages indicate that the problem occurred prior to the build step that's emitting the errors.

**Comment by anthonycastelli** [comment]
That is everything that Xcode is spitting out.

**Comment by bdash** [comment]
That's interesting. What is producing `/Users/Anthony/Library/Developer/Xcode/DerivedData/Shyft-aiciqwatyeujqlazknpolfflpbvx/Build/Products/Debug-iphonesimulator/Shyft.app/Frameworks/Realm.framework/Realm`? Is that a Realm framework you've built yourself? The script phase that is failing appears to expect it to be a fat (universal) binary, but it's not. Why does the script expect it to be a fat binary? Debug builds are typically made for only a single architecture (`ONLY_ACTIVE_ARCH = YES` in Xcode parlance).

**Comment by anthonycastelli** [comment]
Right, I've gone through and wiped out my derived data just incase that was causing issues, but it still persists. I tried building a copy of the frame when Swift 2.2 was released, but I removed all reminance of that.

**Comment by bdash** [comment]
Note that I was not suggesting you change anything about your configuration, but rather simply making observations based on the output you provided. Do you have answers for the questions I raised?

**Comment by anthonycastelli** [comment]
Oh yeah, I knew what you meant. I have a pod for Realm and RealmSwift in this project. Its the only project I am currently using Realm in, so I guess what's producing the framework is Cocoapods.

**Comment by bdash** [comment]
It seems to me that the problem is that your script phase expects Realm to be a universal binary, when it won't be in builds where `ONLY_ACTIVE_ARCH = YES` (i.e., debug builds). If you disable your script phase, do you have better luck? If so, you'll need to fix your script phase's expectations to match reality.

**Comment by anthonycastelli** [comment]
I seemed to have fixed the issue be resetting up cocoa pods with the project. No idea how or what was causing this issue.

**Comment by stel** [comment]
@anthonycastelli could you please provide some instructions that helped you resolve this issue? Thanks!

**Comment by anthonycastelli** [comment]
I just removed cocoapods bother from the project and my system, rebooted and tried again. Seemed to of fixed it and no issues since. I was running a beta version of cocoa pods so something might of been corrupt on my initial install.

**Comment by bdash** [comment]
Pleased to hear that you got this working.
